pkg/evaluator/scope.go

```diff
@@ -29,3 +31,17 @@ func (s *scope) set(name string, val value) {
 	}
 	s.values[name] = val
 }
+
+func (s *scope) update(name string, val value) {
+	if s == nil {
```

Comment:
Rather than doing this check here (which is not done for the other methods), perhaps move the nil check below before calling `s.outer.update()` so you can panic with a slightly better error message.

```
if s.outer == nil {
  panic(fmt.Errorf("%s: update of unknown variable %q", ErrAssignmentTarget, name))
}
```